tests/system/Validation/ValidationTest.php

```diff
@@ -789,6 +790,25 @@ public function testJsonInput(): void
         unset($_SERVER['CONTENT_TYPE']);
     }
 
+    public function testJsonInputInvalid(): void
+    {
+        $this->expectException(HTTPException::class);
+        $this->expectExceptionMessage('Failed to parse JSON string. Error: Syntax error');
+
+        $config  = new App();
+        $json    = 'invalid';
+        $request = new IncomingRequest($config, new URI(), $json, new UserAgent());
+        $request->setHeader('Content-Type', 'application/json');
+
+        $rules = [
+            'role' => 'if_exist|max_length[5]',
+        ];
+        $this->validation
+            ->withRequest($request->withMethod('POST'))
+            ->setRules($rules)
+            ->run();
+    }
+
```

Comment:
Based on my comment above it would be good to add a test or two around proper handling with `JSON_THROW_ON_ERROR`

Reply:
Added a test to show the behavior.